services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/registry.py

```diff
@@ -46,47 +50,94 @@ async def _is_registry_reachable(registry_settings: RegistrySettings) -> None:
+                }
+            )
+
+        # NOTE: uses different URL than the one specified in the configuration
+        url = "https://hub.docker.com/v2/users/login"
+
+        _logger.info("Registry test url ='%s'", url)
+        response = await client.post(url, **params)
+        reachable = (
+            response.status_code == status.HTTP_200_OK and "token" in response.json()
+        )
+        if not reachable:
+            _logger.error("Response: %s %s", response, response.text)
+            error_message = (
+                f"Could not reach registry {response.request.url}"
+                f"auth={registry_settings.REGISTRY_AUTH}, because: {response.text}"
+            )
+            raise _RegistryNotReachableError(error_message)
+
+
+async def wait_for_registries_liveness(app: FastAPI) -> None:
+    settings: ApplicationSettings = app.state.settings
 
     await wait_for_service_liveness(
         _is_registry_reachable,
-        service_name="Registry",
-        endpoint=_get_registry_url(registry_settings),
-        registry_settings=registry_settings,
+        service_name="Internal Registry",
+        endpoint=_get_registry_url(settings.DY_DEPLOYMENT_REGISTRY_SETTINGS),
+        registry_settings=settings.DY_DEPLOYMENT_REGISTRY_SETTINGS,
     )
 
+    if settings.DY_DOCKER_HUB_REGISTRY_SETTINGS:
+        await wait_for_service_liveness(
+            _is_dockerhub_reachable,
+            service_name="DockerHub Registry",
+            endpoint="https://hub.docker.com/v2/users/login",
+            registry_settings=settings.DY_DOCKER_HUB_REGISTRY_SETTINGS,
+        )
 
-async def _login_registry(registry_settings: RegistrySettings) -> None:
+
+async def _login_registries(settings: ApplicationSettings) -> None:
     """
     Creates ~/.docker/config.json and adds docker registry credentials
     """
 
-    def create_docker_config_file(registry_settings: RegistrySettings) -> None:
+    def _get_credentials(registry_settings: RegistrySettings) -> str:
         user = registry_settings.REGISTRY_USER
         password = registry_settings.REGISTRY_PW.get_secret_value()
+        return base64.b64encode(f"{user}:{password}".encode()).decode()
+
+    def create_docker_config_file(registries_settings: list[RegistrySettings]) -> None:
         docker_config = {
             "auths": {
-                f"{registry_settings.resolved_registry_url}": {
-                    "auth": base64.b64encode(f"{user}:{password}".encode()).decode(
-                        "utf-8"
-                    )
+                f"{settings.resolved_registry_url}": {
+                    "auth": _get_credentials(settings)
                 }
+                for settings in registries_settings
             }
         }
-        conf_file = Path.home() / ".docker" / "config.json"
-        conf_file.parent.mkdir(exist_ok=True, parents=True)
-        conf_file.write_text(json.dumps(docker_config))
 
```

Comment:
Actually same here, this can break if the docker version somehow decides to change its usage. If you would run the ```docker login``` instead it would automatically write into the correct file(s)